src/Azure/Azure.Quantum.Client/SubmitterFactory.cs

```diff
@@ -0,0 +1,122 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#nullable enable
+
+namespace Microsoft.Azure.Quantum
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Collections.Immutable;
+    using System.Linq;
+    using System.Reflection;
+    using Microsoft.Quantum.Runtime.Submitters;
+
+    /// <summary>
+    /// Creates Azure Quantum submitters for specific targets.
+    /// </summary>
+    public static class SubmitterFactory
+    {
+        /// <summary>
+        /// Information about each supported QIR submitter.
+        /// </summary>
+        private static readonly ImmutableList<SubmitterInfo> QirSubmitters = ImmutableList.Create(
+            new SubmitterInfo(
+                "microsoft",
```

Comment:
@cesarzc, do you know what the target prefix for MicrosoftSimulatorSubmitter should be? I guessed `microsoft.*`, but maybe it's `microsoft.simulator.*` or something else.

Reply:
Currently it is `microsoft.*`, and for the full state simulator the target name is currently `microsoft.simulator`.

Reply:
This is just the current name that the service team is currently working with. We are coordinating with them to come up with a naming convention that makes sense.